packages/commcare/src/Adaptor.js

```diff
@@ -77,23 +78,62 @@ export function get(path, params = {}, callback = s => s) {
       params
     );
 
+    let { offset = 0, limit = 1000 } = resolvedParams;
+
+    let nextState = state;
+    let result;
+    let allowPagination = !isNaN(resolvedParams.offset);
+
     try {
-      const response = await request(
-        state.configuration,
-        `/a/${domain}/api/v0.5/${resolvedPath}`,
-        {
-          method: 'GET',
-          params: resolvedParams,
-          contentType: 'application/json',
+      let requestParams = {
+        ...resolvedParams,
+      };
+
+      do {
+        const response = await request(
+          state.configuration,
+          `/a/${domain}/api/v0.5/${resolvedPath}`,
+          {
+            method: 'GET',
+            params: requestParams,
+            contentType: 'application/json',
+          }
+        );
+
+        nextState = prepareNextState(state, response, callback);
+
+        if (response?.body?.meta?.next) {
+          if (!result) {
+            result = [];
+          }
+          offset = response?.body?.meta?.offset + response?.body?.meta?.limit;
```

Comment:
We don't need to use the optional chaining operator `?` so much here (and in other lines).  If we get  to this line, we know that `response.body.meta.next` is an object